portfolio/src/main/webapp/business-logic/Test.js

```diff
@@ -0,0 +1,131 @@
+        title: "My first",
+        description: "My desc",
+        imageUrl: "/images/cat1.jpg",
+        labels: [],
+        active: true
+    }, {
+        title: "My second",
+        description: "Lorem Ipsum",
+        imageUrl: "/images/cat2.jpg",
+        labels: [{
+            id: 1,
+            name: "C++",
+            color: "red"
+        }, {
+            id: 2,
+            name: "Java",
+            color: "green"
+        }, {
+            id: 3,
+            name: "Python",
+            color: "black"
+        }, {
+            id: 4,
+            name: "JavaScript",
+            color: "blue"
+        }],
+        active: true
+    }, {
+        title: "My 3rd",
+        description: "Lorem Ipsum",
+        imageUrl: "/images/cat2.jpg",
+        labels: [{
+            id: 1,
+            name: "C++",
+            color: "red"
+        }, {
+            id: 4,
+            name: "JavaScript",
+            color: "blue"
+        }],
+        active: true
+    }, {
+        title: "My 4th",
+        description: "Lorem Ipsum",
+        imageUrl: "/images/cat1.jpg",
+        labels: [{
+            id: 2,
+            name: "Java",
+            color: "green"
+        }, {
+            id: 3,
+            name: "Python",
+            color: "black"
+        }, {
+            id: 4,
+            name: "JavaScript",
+            color: "blue"
+        }],
+        active: true
+    }, {
+        title: "My 5th",
+        description: "Lorem Ipsum",
+        imageUrl: "/images/cat2.jpg",
+        labels: [{
+            id: 3,
+            name: "Python",
+            color: "black"
+        }],
+        active: true
+    }, {
+        title: "My 6th",
+        description: "Lorem Ipsum",
+        imageUrl: "/images/cat1.jpg",
+        labels: [{
+            id: 1,
+            name: "C++",
+            color: "red"
+        }, {
+            id: 3,
+            name: "Python",
+            color: "black"
+        }, {
+            id: 4,
+            name: "JavaScript",
+            color: "blue"
+        }],
+        active: true
+    }, {
+        title: "My 7th",
+        description: "Lorem Ipsum LASTTTT",
+        imageUrl: "/images/cat2.jpg",
+        labels: [],
+        active: true
+    }];
+
+    const labels = [{
+        id: 1,
+        name: "C++",
+        color: "red"
+    }, {
+        id: 2,
+        name: "Java",
+        color: "green"
+    }, {
+        id: 3,
+        name: "Python",
+        color: "black"
+    }, {
+        id: 4,
+        name: "JavaScript",
+        color: "blue"
+    }];
+
+    const filterController = new FilterController(labels, labelListRenderer);
+    const projectController = new ProjectController(projects, filterController, mainContentRenderer);
+    const internalController = new InternalController(projectController, filterController);
+
+    //labelListRenderer.render(labels);
+    //mainContentRenderer.render(projects);
+    internalController.setup();
+}
```

Comment:
But you doesn't use if for (unit) test.

Reply:
I tried to make all the classes to use dependency injection. I guess this could make our life easier. But I guess testing is out of the scope for this week. Maybe, if time's left, I can do that during the 5th week?